server/api/users/users.controller.ts

```diff
@@ -18,15 +18,23 @@ export const getCurrentUser: RequestHandler<void, ClientUser>  = (req, res) => {
 };
 
 type UserProfile = Pick<User, 'id'
-| 'discord_user_id'
-| 'discord_name'
-| 'bio'
-| 'twitter_username'
-| 'linkedin_url'
-| 'github_username'
-| 'website'>
-
-export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req, res) => {
+  | 'discord_user_id'
+  | 'discord_name'
+  | 'bio'
+  | 'twitter_username'
+  | 'linkedin_url'
+  | 'github_username'
+  | 'website'>
+
+interface UpdatableFields {
+  bio?: string;
+  twitter_username?: string;
+  linkedin_url?: string;
+  github_username?: string;
+  webiste?: string;
+}
```

Comment:
for this you can use a combination of built-in typescript `Partial` and `Pick`

also, typo for website :)